staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/strategy_test.go

```diff
@@ -0,0 +1,260 @@
+			statusEnabled: false,
+			old: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"metadata": generation1(),
+					"spec":     "old",
+					"status":   "old",
+				},
+			},
+			obj: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"metadata": generation1(),
+					"spec":     "new",
+					"status":   "old",
+				},
+			},
+			expected: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"metadata": generation2(),
+					"spec":     "new",
+					"status":   "old",
+				},
+			},
+		},
+		{
+			name:          "/status is disabled, status changes increment generation",
+			statusEnabled: false,
+			old: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"metadata": generation1(),
+					"spec":     "old",
+					"status":   "old",
+				},
+			},
+			obj: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"metadata": generation1(),
+					"spec":     "old",
+					"status":   "new",
+				},
+			},
+			expected: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"metadata": generation2(),
+					"spec":     "old",
+					"status":   "new",
+				},
+			},
+		},
+		{
+			name:          "/status is disabled, other top-level field changes increment generation",
+			statusEnabled: false,
+			old: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"metadata": generation1(),
+					"spec":     "old",
+					"image":    "old",
+					"status":   "old",
+				},
+			},
+			obj: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"metadata": generation1(),
+					"spec":     "old",
+					"image":    "new",
+					"status":   "old",
+				},
+			},
+			expected: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"metadata": generation2(),
+					"spec":     "old",
+					"image":    "new",
+					"status":   "old",
+				},
+			},
+		},
+		{
+			name:          "/status is disabled, metadata changes do not increment generation",
+			statusEnabled: false,
+			old: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"metadata": map[string]interface{}{
+						"generation": int64(1),
+						"other":      "old",
+					},
+					"spec":   "old",
+					"status": "old",
+				},
+			},
+			obj: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"metadata": map[string]interface{}{
+						"generation": int64(1),
+						"other":      "new",
+					},
+					"spec":   "old",
+					"status": "old",
+				},
+			},
+			expected: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"metadata": map[string]interface{}{
+						"generation": int64(1),
+						"other":      "new",
+					},
+					"spec":   "old",
+					"status": "old",
+				},
+			},
+		},
+	}
+	defer utilfeaturetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, apiextensionsfeatures.CustomResourceSubresources, true)()
```

Comment:
Come to think of it, we don't need this. The feature gate is already enabled by default since it's beta.